omnisafe/adapter/modelbased_adapter.py

```diff
@@ -0,0 +1,382 @@
+
+    :class:`ModelBasedAdapter` is used to adapt the environment to the model-based training.
+
+
+    Args:
+        env_id (str): The environment id.
+        num_envs (int): The number of environments.
+        seed (int): The random seed.
+        cfgs (Config): The configuration.
+        kwargs(dict): The other keyword arguments.
+
+    Attributes:
+        _env_id (str): The environment id.
+        _device (torch.device): The device.
+        _env (CMDP): The environment.
+        _cfgs (Config): The configuration.
+        _ep_ret (torch.Tensor): The episode return.
+        _ep_cost (torch.Tensor): The episode cost.
+        _ep_len (torch.Tensor): The episode length.
+        _last_dynamics_update (float): The last time of dynamics update.
+        _last_policy_update (float): The last time of policy update.
+        _last_eval (float): The last time of evaluation.
+        coordinate_observation_space (OmnisafeSpace): The coordinate observation space.
+        lidar_observation_space (OmnisafeSpace): The lidar observation space.
+        task (str): The task. eg. The task of SafetyPointGoal-v0 is 'goal'
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        env_id: str,
+        num_envs: int,
+        seed: int,
+        cfgs: Config,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the model-based adapter."""
+        assert env_id in support_envs(), f'Env {env_id} is not supported.'
+
+        self._env_id = env_id
+        self._device = cfgs.train_cfgs.device
+
+        self._env = make(env_id, num_envs=num_envs, device=cfgs.train_cfgs.device, **kwargs)
+
+        # wrap the environment, use the action repeat in model-based setting.
+        self._wrapper(
+            obs_normalize=cfgs.algo_cfgs.obs_normalize,
+            reward_normalize=cfgs.algo_cfgs.reward_normalize,
+            cost_normalize=cfgs.algo_cfgs.cost_normalize,
+            action_repeat=cfgs.algo_cfgs.action_repeat,
+        )
+        self._env.set_seed(seed)
+        self._cfgs = cfgs
+        if hasattr(self._env, 'coordinate_observation_space') and hasattr(
+            self._env,
+            'lidar_observation_space',
+        ):
+            self.coordinate_observation_space = self._env.coordinate_observation_space
+            self.lidar_observation_space = self._env.lidar_observation_space
+        else:
+            self.coordinate_observation_space = None
+            self.lidar_observation_space = None
+        if hasattr(self._env, 'task'):
+            self.task = self._env.task
+        else:
+            self.task = None
+
+        self._ep_ret: torch.Tensor
+        self._ep_cost: torch.Tensor
+        self._ep_len: torch.Tensor
+        self._current_obs, _ = self.reset()
+        self._max_ep_len = 1000
+        self._reset_log()
+        self._last_dynamics_update = 0
+        self._last_policy_update = 0
+        self._last_eval = 0
+        self._first_log = False
```

Comment:
```suggestion
        self._first_log: bool = False
```
The same with others.